doc/api/stream.md

```diff
@@ -432,6 +432,19 @@ file.end('world!');
 // Writing more now is not allowed!
 ```
 
+[`stream.end()`][stream-end] will call it's callback and emit `'error'`
+with (in order of precedence):
+* `ERR_STREAM_WRITE_AFTER_END` if `chunk` is not nully and
+[`stream.end()`][stream-end] has been called.
+* `ERR_STREAM_DESTROYED` if `chunk` is not nully and
+[`stream.destroy()`][writable-destroy] has been called.
+* `ERR_INVALID_ARG_TYPE` if `chunk` is not nully and
+`chunk` is not a `string` when not in `objectMode`.
```

Comment:
I believe the chunk can also be a `Uint8Array` or `Buffer`. Perhaps we should just link to the `write` method? (``` if the `chunk` is not nully and is of an invalid type for [`stream.write()`][] according to the current mode ```)

Reply:
This is the code:

```js
  if (chunk === null) {
    er = new ERR_STREAM_NULL_VALUES();
  } else if (typeof chunk !== 'string' && !state.objectMode) {
    er = new ERR_INVALID_ARG_TYPE('chunk', ['string', 'Buffer'], chunk);
  }
````

I've tried to put it directly into words.

Reply:
Oh, yeah, this is the **error** flow, and the buffer-ly chunk will lead to `validChunk` not being called at all.